nighthawk/test/options_test.cc

```diff
@@ -0,0 +1,171 @@
+#include "gtest/gtest.h"
+
+#include "test/test_common/utility.h"
+
+#include "nighthawk/source/client/options_impl.h"
+
+using namespace std::chrono_literals;
+
+namespace Nighthawk {
+namespace Client {
+
+class OptionsImplTest : public testing::Test {
+public:
+  OptionsImplTest()
+      : client_name_("nighthawk_client"), good_test_uri_("http://127.0.0.1/"),
+        no_arg_match_("Couldn't find match for argument") {}
+
+  std::unique_ptr<OptionsImpl> createOptionsImpl(const std::string& args) {
+    std::vector<std::string> words = Envoy::TestUtility::split(args, ' ');
+    std::vector<const char*> argv;
+    for (const std::string& s : words) {
+      argv.push_back(s.c_str());
+    }
+    return std::make_unique<OptionsImpl>(argv.size(), argv.data());
+  }
+
+  std::string client_name_;
+  std::string good_test_uri_;
+  std::string no_arg_match_;
+};
+
+TEST_F(OptionsImplTest, BogusInput) {
+  // TODO(oschaaf): enable this test after we have host name validation.
+  // EXPECT_THROW_WITH_REGEX(createOptionsImpl(fmt::format("{} --foo", client_name_)),
+  //                        MalformedArgvException, no_arg_match_);
```

Comment:
?

Reply:
Added a more elaborate comment here. 
```
  // TODO(oschaaf): enable this test after we have host name validation.
  // When just passing the non-existing argument --foo it would be interpreted as a
  // hostname. However, hostnames shouldn't start with '-', and hence this test should
  // not pass.
  // EXPECT_THROW_WITH_REGEX(createOptionsImpl(fmt::format("{} --foo", client_name_)),
  //                        MalformedArgvException, no_arg_match_);
```